File: examples/testng/testng-maven/src/test/java/org/example/ParametrisedTests.java
package org.example;

import org.testng.annotations.DataProvider;
import org.testng.annotations.Parameters;
import org.testng.annotations.Test;

public class ParametrisedTests {
    @DataProvider(name = "data-provider")
    public Object[][] dataProviderMethod() {
        return new Object[][] { { "data one" }, { "data two" }, { "data three" } };
    }

    @Test(dataProvider = "data-provider")
    @Parameters("data")
    public void testMethod_success(String data) {
        System.out.println("Data is: " + data);
    }

    @Test(dataProvider = "data-provider")
    @Parameters("data")
    public void testMethod_failed(String data) {
        System.out.println("Data is: " + data);
        throw new RuntimeException("Test failed");
    }
}

File: examples/kotlin/testng/src/test/kotlin/ParametrisedTests.kt
import org.testng.annotations.DataProvider
import org.testng.annotations.Parameters
import org.testng.annotations.Test

class ParametrisedTests {

    @DataProvider(name = "data-provider")
    fun dataProviderMethod(): Array<Array<Any>> {
        return arrayOf(
            arrayOf("data one"),
            arrayOf("data two"),
            arrayOf("data three")
        )
    }

    @Test(dataProvider = "data-provider")
    @Parameters("data")
    fun testMethod_success1(data: String) {
        println("Data is: $data")
    }

    @Test(dataProvider = "data-provider")
    @Parameters("data")
    fun testMethod_failed1(data: String) {
        println("Data is: $data")
        throw RuntimeException("Test failed")
    }
}
